terraform import fails for resources within a client

Fantastic provider to manage Keycloak btw! I appreciate the thought and effort you've put into it.

Similar to https://github.com/mrparkers/terraform-provider-keycloak/issues/399; but I've confirmed login works for me, but it fails further down the chain.

I'm running into an error when importing a bunch of existing resources original created outside Terraform. It appears wide-spread across resources, permissions, etc perhaps because they share very similar conventions. So far I've only had luck importing a Client.

My command for the import

```bash
$ TF_LOG=debug terraform import keycloak_openid_client_authorization_resource.catalog_hive default/
286ebfcf-e924-43ad-9408-53c0526ed76b/25cece67-6a1c-4d33-bcf0-db9c4eddc044
```

Snippet from the logs when I run the import

```txt
2022-11-08T02:48:56.728Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: Login request: tf_req_id=58427d42-11ef-ecef-d906-377fd4a6def5 @caller=github.com/mrparkers/terraform-provider-keycloak/keycloak/keycloak_client.go:108 @module=provider request=client_id=admin-cli&grant_type=password&password=${REDACTED}&username=${REDACTED} tf_provider_addr=provider tf_rpc=Configure timestamp=2022-11-08T02:48:56.727Z
b.com/mrparkers/terraform-provider-keycloak/keycloak/keycloak_client.go:139 tf_provider_addr=provider tf_req_id=58427d42-11ef-ecef-d906-377fd4a6def5 tf_rpc=Configure @module=provider timestamp=2022-11-08T02:48:57.131Z
2022-11-08T02:48:57.131Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: Sending request: @caller=github.com/mrparkers/terraform-provider-keycloak/keycloak/keycloak_client.go:294 tf_req_id=58427d42-11ef-ecef-d906-377fd4a6def5 tf_rpc=Configure @module=provider method=GET path=/auth/admin/serverinfo tf_provider_addr=provider timestamp=2022-11-08T02:48:57.131Z
2022-11-08T02:48:57.662Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: Received response: tf_provider_addr=provider tf_rpc=Configure @module=provider status="200 OK" tf_req_id=58427d42-11ef-ecef-d906-377fd4a6def5 @caller=github.com/mrparkers/terraform-provider-keycloak/keycloak/keycloak_client.go:341 timestamp=2022-11-08T02:48:57.661Z
[0m[1mkeycloak_openid_client_authorization_resource.catalog_hive: Importing from ID "default/286ebfcf-e924-43ad-9408-53c0526ed76b/25cece67-6a1c-4d33-bcf0-db9c4eddc044"...[0m
[31m
[1m[31mError: [0m[0m[1mrpc error: code = Canceled desc = context canceled[0m

[0m[0m[0m
2022-11-08T02:48:57.669Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: 
2022-11-08T02:48:57.669Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: goroutine 60 [running]:
2022-11-08T02:48:57.669Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: github.com/mrparkers/terraform-provider-keycloak/provider.resourceKeycloakOpenidClientAuthorizationResourceImport(0xfa2f00, 0xc000598f30, 0xc0003e3b80, 0xe757a0, 0xc00030ab40, 0x12, 0x0, 0x0, 0x0, 0x20)
2022-11-08T02:48:57.669Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: 	github.com/mrparkers/terraform-provider-keycloak/provider/resource_keycloak_openid_client_authorization_resource.go:190 +0x26d
2022-11-08T02:48:57.669Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ImportState(0xc000114300, 0xfa2f00, 0xc000598f30, 0xc000489770, 0xc00039c4e0, 0x51, 0xc000000001, 0xe82c69, 0x12, 0x0, ...)
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/provider.go:385 +0x544
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ImportResourceState(0xc00000ce70, 0xfa2e58, 0xc000598f30, 0xc0005303e0, 0xe82c69, 0x12, 0x0)
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/grpc_provider.go:1088 +0x11a
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ImportResourceState(0xc00038a500, 0xfa2f00, 0xc000598330, 0xc000318d20, 0x0, 0x0, 0x0)
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: 	github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/tf5server/server.go:841 +0x2ed
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ImportResourceState_Handler(0xe3d4e0, 0xc00038a500, 0xfa2f00, 0xc000598330, 0xc00064e1e0, 0x0, 0xfa2f00, 0xc000598330, 0xc000446000, 0x82)
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: 	github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:403 +0x214
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00032a540, 0xfaa438, 0xc0002d61a0, 0xc0000a8480, 0xc0001edd40, 0x14834d8, 0x0, 0x0, 0x0)
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: 	google.golang.org/grpc@v1.45.0/server.go:1282 +0x52b
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: google.golang.org/grpc.(*Server).handleStream(0xc00032a540, 0xfaa438, 0xc0002d61a0, 0xc0000a8480, 0x0)
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: 	google.golang.org/grpc@v1.45.0/server.go:1619 +0xd0c
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000310040, 0xc00032a540, 0xfaa438, 0xc0002d61a0, 0xc0000a8480)
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: 	google.golang.org/grpc@v1.45.0/server.go:921 +0xab
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: created by google.golang.org/grpc.(*Server).serveStreams.func1
2022-11-08T02:48:57.670Z [DEBUG] plugin.terraform-provider-keycloak_v3.10.0: 	google.golang.org/grpc@v1.45.0/server.go:919 +0x1fd
2022-11-08T02:48:57.672Z [DEBUG] plugin: plugin process exited: path=/root/.terraform.d/plugins/terraform-provider-keycloak_v3.10.0 pid=4259 error="exit status 2"
2022/11/08 02:48:57 [ERROR] <root>: eval: *terraform.EvalImportState, err: rpc error: code = Canceled desc = context canceled
2022/11/08 02:48:57 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Canceled desc = context canceled
```

When I opened the source-code for the specific provider resource import function, [here](https://github.com/mrparkers/terraform-provider-keycloak/blob/v3.10.0/provider/resource_keycloak_openid_client_authorization_resource.go#L183-L193)

I see the following, where it appears we're attempting to access an out of bounds element.

```go
func resourceKeycloakOpenidClientAuthorizationResourceImport(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
	parts := strings.Split(d.Id(), "/")
	if len(parts) != 3 {
		return nil, fmt.Errorf("Invalid import. Supported import formats: {{realmId}}/{{resourceServerId}}/{{authorizationResourceId}}")
	}
	d.Set("realm_id", parts[0])
	d.Set("resource_server_id", parts[1])
	d.SetId(parts[3]) // <----- Shouldn't this be d.SetId(parts[2]) ?

	return []*schema.ResourceData{d}, nil
}
```

Could you please advice me on whether I'm doing something wrong with my import/configuration, or if there's a legitimate issue with the provider?

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -42,3 +42,10 @@ services:
 # Make the custom-user-federation-example extension available to Keycloak. The :z option is required and tells Docker that the volume content will be shared between containers.
     - ./custom-user-federation-example/build/libs/custom-user-federation-example.jar:/opt/jboss/keycloak/standalone/deployments/custom-user-federation-example.jar:z
 
+  app:
+    image: golang:1.18
+    profiles:
+      - build
+    volumes:
+      - .:/app
+    working_dir: /app

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -41,11 +41,3 @@ services:
     volumes:
 # Make the custom-user-federation-example extension available to Keycloak. The :z option is required and tells Docker that the volume content will be shared between containers.
     - ./custom-user-federation-example/build/libs/custom-user-federation-example.jar:/opt/jboss/keycloak/standalone/deployments/custom-user-federation-example.jar:z
-
-  app:
-    image: golang:1.18
-    profiles:
-      - build
-    volumes:
-      - .:/app
-    working_dir: /app

--- a/makefile
+++ b/makefile
@@ -4,8 +4,10 @@ GOARCH?=amd64
 
 MAKEFLAGS += --silent
 
+VERSION=$$(git describe --tags)
+
 build:
-	go build -o terraform-provider-keycloak
+	CGO_ENABLED=0 go build -trimpath -ldflags "-s -w -X main.version=$(VERSION)" -o terraform-provider-keycloak_$(VERSION)
 
 build-example: build
 	mkdir -p example/.terraform/plugins/terraform.local/mrparkers/keycloak/4.0.0/$(GOOS)_$(GOARCH)

--- a/provider/resource_keycloak_openid_client_authorization_permission.go
+++ b/provider/resource_keycloak_openid_client_authorization_permission.go
@@ -195,7 +195,7 @@ func resourceKeycloakOpenidClientAuthorizationPermissionImport(_ context.Context
 	}
 	d.Set("realm_id", parts[0])
 	d.Set("resource_server_id", parts[1])
-	d.SetId(parts[3])
+	d.SetId(parts[2])
 
 	return []*schema.ResourceData{d}, nil
 }

--- a/provider/resource_keycloak_openid_client_authorization_resource.go
+++ b/provider/resource_keycloak_openid_client_authorization_resource.go
@@ -187,7 +187,7 @@ func resourceKeycloakOpenidClientAuthorizationResourceImport(_ context.Context, 
 	}
 	d.Set("realm_id", parts[0])
 	d.Set("resource_server_id", parts[1])
-	d.SetId(parts[3])
+	d.SetId(parts[2])
 
 	return []*schema.ResourceData{d}, nil
 }

--- a/provider/resource_keycloak_openid_client_authorization_scope.go
+++ b/provider/resource_keycloak_openid_client_authorization_scope.go
@@ -130,7 +130,7 @@ func resourceKeycloakOpenidClientAuthorizationScopeImport(_ context.Context, d *
 	}
 	d.Set("realm_id", parts[0])
 	d.Set("resource_server_id", parts[1])
-	d.SetId(parts[3])
+	d.SetId(parts[2])
 
 	return []*schema.ResourceData{d}, nil
 }

